homeassistant/components/utility_meter/sensor.py

```diff
@@ -243,9 +238,7 @@ def async_reading(self, event):
     @callback
     def async_tariff_change(self, event):
         """Handle tariff changes."""
-        if (new_state := event.data.get("new_state")) is None:
-            return
-
+        new_state = event.data.get("new_state")
```

Comment:
I'm not sure we can remove the guard above. If an entity is removed we fire an event with `"new_state"` event data key with value `None`.

https://github.com/home-assistant/core/blob/ffe821a1f7602a5c3e5bc4ca395358d4838873ff/homeassistant/core.py#L1266-L1271

Reply:
overlooked that situation :(